Updates relating to macOS support

lbarry316 opened the pull request:
## Description

As requested by Ankur Mundhra

## Definition of Done

## Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussion that's relevant]

## Related PRs

## Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

lbarry316 committed:
Removed content relating to macOS 12 and below

Signed-off-by: LisaBarry316 <lisa.barry@progress.com>

AnkurMundhra reviewed (commented):
Let us not change it for current release of Chef 18. Please work with Rahul to confirm when we need to update, most probably in next release of Client 18.

lbarry316 committed:
Updates based on input from Ankur Mundhra

Signed-off-by: Ian Maddaus <ian.maddaus@progress.com>

lbarry316 commented:
> Let us not change it for current release of Chef 18. Please work with Rahul to confirm when we need to update, most probably in next release of Client 18.

As mentioned in Teams:
"I made updates in https://github.com/chef/chef-web-docs/pull/4486 - Deploy Preview: [https://deploy-preview-3430--chef-workstation.netlify.app/platforms/](https://deploy-preview-4486--chef-web-docs.netlify.app/platforms/) [:](https://deploy-preview-3430--chef-workstation.netlify.app/platforms/)
- Added back in the macOS 12.x content in the Commercial support section for Chef Infra Client:
<img width="944" height="49" alt="image" src="https://github.com/user-attachments/assets/dca61945-5274-4fb7-a59c-e672c99e0aaf" />
- Updated the Platform end-of-life policy as follows:
<img width="1417" height="43" alt="image" src="https://github.com/user-attachments/assets/36785d1a-1689-46b8-b61b-af27b834e4f8" />
The 2024 date relates to the **Vendor End-of Life Date** (not Chef's)
So I think the changes (which are just improvements now) in might be OK to merge now, but let us know"

However, if you want us to simply close this pull request and not merge it, let us know
